packages/mui-base/src/utils/appendOwnerState.ts

```diff
@@ -38,11 +38,11 @@ export default function appendOwnerState<
   OtherProps extends Record<string, any>,
   OwnerState,
 >(
-  elementType: ElementType,
+  elementType: ElementType | undefined,
   otherProps: OtherProps,
   ownerState: OwnerState,
 ): AppendOwnerStateReturnType<ElementType, OtherProps, OwnerState> {
-  if (isHostComponent(elementType)) {
+  if (elementType && isHostComponent(elementType)) {
```

Comment:
There's no need to append ownerState when ElementType is not defined.

```suggestion
  if (elementType === undefined || isHostComponent(elementType)) {
```

Reply:
Thanks for the quick review! My bad, fixed it.